app/lib/widgets/CrossSigning.dart

```diff
@@ -734,61 +739,74 @@ class CrossSigning {
           ),
         ),
         const SizedBox(height: 5),
-        waitForMatch
-            ? Center(
-                child: Padding(
-                  padding: const EdgeInsets.all(10),
-                  child: Text(
-                    waitingFor,
-                    style: CrossSigningSheetTheme.secondaryTextStyle,
-                  ),
-                ),
-              )
-            : Row(
-                mainAxisAlignment: MainAxisAlignment.center,
-                children: [
-                  Container(
-                    padding: const EdgeInsets.only(left: 20),
-                    width: MediaQuery.of(context).size.width * 0.48,
-                    child: elevatedButton(
-                      AppLocalizations.of(context)!.verificationSasDoNotMatch,
-                      CrossSigningSheetTheme.redButtonColor,
-                      () async {
-                        // mismatch sas verification
-                        await event.mismatchSasVerification();
-                        // go to onCancel status
-                        Get.back();
-                        Future.delayed(const Duration(milliseconds: 500), () {
-                          _onKeyVerificationCancel(event, true);
-                        });
-                      },
-                      CrossSigningSheetTheme.buttonTextStyle,
-                    ),
-                  ),
-                  const SizedBox(width: 5),
-                  Container(
-                    padding: const EdgeInsets.only(right: 20),
-                    width: MediaQuery.of(context).size.width * 0.48,
-                    child: elevatedButton(
-                      AppLocalizations.of(context)!.verificationSasMatch,
-                      CrossSigningSheetTheme.greenButtonColor,
-                      () async {
-                        setState(() => waitForMatch = true);
-                        // confirm sas verification
-                        await event.confirmSasVerification();
-                        // close dialog
-                        setState(() => waitForMatch = false);
-                        // go to onMac status
-                        Get.back();
-                        Future.delayed(const Duration(milliseconds: 500), () {
-                          _onKeyVerificationMac(event);
-                        });
-                      },
-                      CrossSigningSheetTheme.buttonTextStyle,
-                    ),
-                  ),
-                ],
-              ),
+        _buildBodyOnKey(context, event, setState),
+      ],
+    );
+  }
+
+  Widget _buildBodyOnKey(
+    BuildContext context,
+    VerificationEvent event,
+    Function setState,
+  ) {
+    if (waitForMatch) {
+      String waitingFor = sprintf(
```

Comment:
Is the string interpolation necessary here?

Reply:
Fixed it